rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,196 @@
+        At the moment it does not decode the fee in ETH. For example:
+        etherscan.io/tx/0xeeea20b39f157fe59fa4904fd4b62f8971188b53d05c6831e0ed67ee157e40c2"""
+        if not self.base.is_tracked(sender) and not self.base.is_tracked(receiver):
+            return DEFAULT_DECODING_OUTPUT
+
+        out_event = in_event = returned_event = None
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND or
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == sender
+            ):  # find the send event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.address = self.router_address
+                out_event = event
+            elif event.event_type == HistoryEventType.RECEIVE and event.location_label == sender:  # find the receive event  # noqa: E501
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.counterparty = CPT_PARASWAP
+                event.notes = f'Receive {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} as the result of a swap in paraswap'  # noqa: E501
+                event.address = self.router_address
+                # use this index as the event may be a native currency transfer
+                # and appear at the start
+                event.sequence_index = context.tx_log.log_index
+                if in_event is None:
+                    in_event = event
+                else:  # if another in_event is found, then it is the return of in_asset
+                    returned_event = event  # type: ignore[unreachable]
+
+        if in_event is None or out_event is None:
+            self.log.error('Could not find the corresponding events for decoding paraswap swap')
+            return DEFAULT_DECODING_OUTPUT
+
+        if returned_event is not None:
+            out_event.balance.amount -= returned_event.balance.amount  # type: ignore[unreachable]
+            context.decoded_events.remove(returned_event)
+        out_event.notes = f'Swap {out_event.balance.amount} {out_event.asset.resolve_to_asset_with_symbol().symbol} in paraswap'  # noqa: E501
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
+
+        fee_raw = fee_asset = None  # extract the fee info
+        for log in context.all_logs:
+            if (
+                log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(log.topics[2]) == self.fee_receiver_address
```

Comment:
the fee receiver address is only used to receive fees?

Reply:
Yes, its etherscan name is also [Paraswap v5: Fee Claimer](https://etherscan.io/address/0xeF13101C5bbD737cFb2bF00Bbd38c626AD6952F7)